### ISSUE TITLE
On Android M, app crashes when using emulator

### ISSUE THREAD

**Issue by kyletan7** [ArthurHub/Android-Image-Cropper#107]
After getting permissions on Android M for getting image via camera, it crashes after trying to send the captured picture back to my activity. It looks like this is because the camera activity does not return anything in Intent. Below is the crash log. 

`                                                                                  java.lang.RuntimeException: Performing stop of activity that is not resumed: {com.example.kyle.wallpaperpicker/com.example.kyle.wallpaperpicker.MainActivity}
                                                                                      at android.app.ActivityThread.performStopActivityInner(ActivityThread.java:3450)
                                                                                      at android.app.ActivityThread.handleStopActivity(ActivityThread.java:3531)
                                                                                      at android.app.ActivityThread.access$1100(ActivityThread.java:156)
                                                                                      at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1380)
                                                                                      at android.os.Handler.dispatchMessage(Handler.java:102)
                                                                                      at android.os.Looper.loop(Looper.java:211)
                                                                                      at android.app.ActivityThread.main(ActivityThread.java:5373)
                                                                                      at java.lang.reflect.Method.invoke(Native Method)
                                                                                      at java.lang.reflect.Method.invoke(Method.java:372)
                                                                                      at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1020)
                                                                                      at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:815)`

I am using this to launch the camera intent:

`                    Intent cameraIntent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
                    startActivityForResult(cameraIntent, CAMERA_IMAGE_REQUEST);`

and to read data from the callback: 
`                    Uri cameraImageUri = data.getData();
`

Also in the FAQ part for `onRequestPermissionsResult` I think there has to be a check for the permissions code for accessing internal storage by including:

`        if (requestCode == CropImage.PICK_IMAGE_PERMISSIONS_REQUEST_CODE) 
`

**Comment by ArthurHub** [comment]
Can you reproduce it in the sample app? It uses all the permissions check needed and uses internal URI for camera so no external storage is required.
See what is the different between your code and the sample app code or the code for creating the pick-from-camera-intent.
Also it may be an issue with the emulator, I remember having the camera crash on specific emulator no matter what I did.

fixed the wiki, thx.

**Comment by kyletan7** [comment]
I will try it on the sample app shortly. 
I tried running on API21 and 22 on an emulator and it also caused a crash, so it isn't specific to Android M or related to permissions I guess. 
A related question I found: http://stackoverflow.com/questions/24503968/camera-intent-not-returning-to-calling-activity

P.S. Running it on an actual device worked perfectly on the other hand. So it might be like you said - an emulator specific problem.

**Comment by kyletan7** [comment]
@ArthurHub Solved the issue. It crashes because on the emulator the camera intent returns null as explained in the SO question. Turns out you figured an elegant and quick way to solve this as shown in `getPickImageResultUri`. I'm separating the camera and gallery intents instead of combining everything into one like you did, so I just reused some of the code in there and it worked like a charm. Thank you for the excellent library!
